src/CMakeLists.txt

```diff
@@ -208,6 +206,11 @@ IF(LMMS_BUILD_WIN32)
 	SET_TARGET_PROPERTIES(lmms PROPERTIES
 		ENABLE_EXPORTS ON
 	)
+	IF(NOT MSVC)
+		SET_PROPERTY(TARGET lmms
+			APPEND_STRING PROPERTY LINK_FLAGS " -mwindows"
+		)
```

Comment:
Why don't you use `WIN32_EXECUTABLE` target property instead?

Reply:
From my original PR comment:
> I'm not using CMake's WIN32_EXECUTABLE property yet, since that requires extra jiggery-pokery for MSVC which is out of scope here.

Specifically:
* We're currently using `-mwindows`; switching to `WIN32_EXECUTABLE` would be an additional change which isn't the purpose of this PR.
* This would require setting a custom entry point for MSVC. We need to work on Windows program startup in general, e.g. adding proper Unicode handling for command line arguments, and I think these would be better dealt with together.
* A personal point, but I quite like that my debug builds open a console window when launched. I'd rather set this based on configuration (i.e. console for debug, none for release, not sure about RelWithDebInfo), which complicates things further.